helix-core/src/syntax.rs

```diff
@@ -293,9 +302,25 @@ impl Loader {
                     .and_then(|extension| self.language_config_ids_by_file_type.get(extension))
             });
 
-        configuration_id.and_then(|&id| self.language_configs.get(id).cloned())
+        // If we have not found the configuration_id, see if we can get it from a shebang line
+        if configuration_id.is_none() {
+            if let Ok(mut file) = File::open(path) {
+                let mut buf = [0; 100];
+                if std::io::Read::read(&mut file, &mut buf[..]).is_ok() {
```

Comment:
You need to use the return value from `read` to index into `&buf[..n]`. `read` has no guarantees on how many bytes it will read.

Reply:
Better yet, this should be `let n = file.take(128).read_to_end(&mut buf)?;`

Reply:
And to get rid of the `str::from_utf8` and two whole levels of indentation:

```rust
let file = File::open(path).ok()?;
let buf = String::with_capacity(128);
file.take(128).read_to_string(&mut buf).ok()?;
```